lib/msf/jmx/discovery.rb

```diff
@@ -0,0 +1,26 @@
+# -*- coding: binary -*-
+
+module Msf
+  module Jmx
+    # This module provides methods which help to handle JMX end points discovery
+    module Discovery
+      # Builds a Rex::Java::Serialization::Model::Stream to discover
+      # an JMX RMI endpoint
+      #
+      # @return [Rex::Java::Serialization::Model::Stream]
+      def discovery_stream
+        block_data = Rex::Java::Serialization::Model::BlockData.new(
+          nil,
+          "\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00" +
+              "\x00\x00\x00\x02\x44\x15\x4d\xc9\xd4\xe6\x3b\xdf"
```

Comment:
Here and elsewhere where you appear to be building the building blocks of the protocol using these hex encoded strings, it would be very helpful from a review/etc perspective if you could add some commentary to them.  IMO, when we are adding new protocol support to something used in side of Metasploit, we should do our best to ensure that we document that protocol as much as possible.  Otherwise this looks like magic.